django/db/models/sql/compiler.py

```diff
@@ -547,7 +547,9 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
                 # to exclude extraneous selects.
                 sub_selects = []
                 sub_params = []
-                for select, _, alias in self.select:
+                for index, (select, _, alias) in enumerate(self.select, start=1):
+                    if not alias and with_col_aliases:
+                        alias = 'Col%d' % index
```

Comment:
Converting `sub_params + params` to a `tuple` uncovered this issue with the mishandling of `with_col_aliases`.